pkgs/test_api/lib/src/backend/util/identifier_regex.dart

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// A regular expression matching a full string as a hyphenated identifier.
+///
+/// This is like a standard Dart identifier, except that it can also contain
+/// hyphens.
+final anchoredHyphenatedIdentifier = RegExp(r'^[a-zA-Z_-][a-zA-Z0-9_-]*$');
```

Comment:
Where is this used?

Reply:
https://github.com/dart-lang/test/blob/e779df6cc280104d0ce119c202417cd62b20438f/pkgs/test_core/lib/src/runner/parse_metadata.dart#L190

https://github.com/dart-lang/test/blob/e779df6cc280104d0ce119c202417cd62b20438f/pkgs/test_core/lib/src/runner/configuration/load.dart#L540

https://github.com/dart-lang/test/blob/e779df6cc280104d0ce119c202417cd62b20438f/pkgs/test_api/lib/src/backend/metadata.dart#L288